clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -322,6 +322,16 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                                 TRACE_MARKER_TYPE_SYSCALL,
                         "Syscall instruction not followed by syscall marker");
     }
+    if (memref.marker.type == TRACE_TYPE_MARKER &&
+        memref.marker.marker_type == TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL) {
+        shard->found_blocking_marker_ = true;
+        report_if_false(shard,
+                        shard->found_syscall_marker_ &&
```

Comment:
`shard->found_syscall_marker_ &&` can be omitted I think, since that would be set if the second condition is true.

Reply:
Done.